### PR TITLE
[Notification] Send notification APIs

### PR DESCRIPTION
**Pull request by qnkhuat**
Epic: https://github.com/metabase/metabase/issues/50766
Add 2 apis:
- POST /api/notification/:id/send used to trigger sending a saved notification, can also take handler_ids list to selectively send to a list of handlers
- POST /api/notification/send used to trigger sending an unsaved notification

Note: we don't have permission checks yet, I'll revisit all the API permission model later, need to check with Product to set our permission model for notification.

### CODE REVIEW

File: src/metabase/api/notification.clj
Comment (qnkhuat) [inline src/metabase/api/notification.clj:1]: man, I'm in love with this api namespace so much, it's so _simple_

File: test/metabase/api/notification_test.clj [hunk 1]
```diff
@@ -65,29 +65,29 @@
...
-                                             :recipients   [{:type    "notification-recipient/user"
-                                                             :user_id (mt/user->id :crowberto)}]}]}]
-          (is (=? (assoc notification :id (mt/malli=? int?))
-                  (mt/user-http-request :crowberto :post 200 "notification" notification))))))
-
-    (testing "card notification with no subscriptions and handler is ok"
-      (let [notification {:payload_type  "notification/card"
-                          :active        true
-                          :creator_id    (mt/user->id :crowberto)
-                          :payload       {:card_id (-> (t2/select-one :model/Card) :id)}}]
```
Comment (qnkhuat) [inline test/metabase/api/notification_test.clj:88]: replaced this card_id with the card-id created by the with temp

File: src/metabase/api/notification.clj
Comment (piranha) [reply src/metabase/api/notification.clj:1]: It's fantastic for sure, hopefully corner cases won't spoil the fun :)

**Review by sansolovyov — APPROVED** [review]
👍

**Pull request merged**